tests/ui/consts/const-eval/transmute-size-mismatch.rs

```diff
@@ -10,15 +10,13 @@ const unsafe fn mir_transmute<T, U>(x: T) -> U {
     mir!{
         {
             RET = CastTransmute(x);
```

Comment:
This should also still check that a note was emitted here.

(You can check for notes with `//~ test` instead of `//~ NOTE: test`; that avoids having to annotate *all* notes.)